pennylane/_queuing.py

```diff
@@ -57,7 +57,9 @@ def __enter__(self):
 
     def __exit__(self, exception_type, exception_value, traceback):
         """Remove this instance from the global list of active contexts."""
-        QueuingContext._active_contexts.remove(self)
+        if QueuingContext._active_contexts[-1] is not self:
+            raise AssertionError("Given QueuingContext is not the current one.")
```

Comment:
I'm curious behind the addition of this validation, is it needed due to different behaviour between `remove` and `pop`?

I'm wondering if we can remove it for even more speedup.

Reply:
It's a structural check, so if this error is ever hit we have bigger problems. You're right that I only added it because of the difference, but since all test pass without it ever triggering I'll remove it.

Reply:
In that case, why not leave it but replace it with an `assert`?

Common practice tends to be to use `assert` for internal consistency checks, and they can be optionally stripped at runtime with `-O` if needed.

Reply:
good idea.